virttest/utils_misc.py

```diff
@@ -4139,7 +4141,7 @@ def __init__(self, command, stdout_tee=None, stderr_tee=None, verbose=True,
 
         # Allow for easy stdin input by string, we'll let subprocess create
         # a pipe for stdin input and we'll write to it in the wait loop
-        if isinstance(stdin, basestring):
+        if isinstance(stdin, string_types):
```

Comment:
If stdin is byte, it will fail. I saw some usage with "basestring = (str,bytes)" and not use six func. @chunfuwen @hsmj1412  please help to confirm, thanks

Reply:
One way follow as:
```
try:
      basestring
except NameError:
      basestring = (str, bytes)
```

Reply:
Below change can match expectation. For u'a', b'a' and 'a' both in Python 2 or Python 3 are all type of str or bytes.
```
if isinstance(str(stdin), (str, bytes)):
```

Reply:
I think chwen's suggestion should be better as str(unicode) should have some problem?

Reply:
Yes, I just modified according to your suggestion.